sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -2118,8 +2113,8 @@ public TimeDomain timeDomain(DoFn<InputT, OutputT> doFn) {
     }
 
     @Override
-    public K key() {
-      return key;
+    public Object key() {
```

Comment:
It seems a regression to not return the typed key. Can OnTimerContext still be typed (necessitating perhaps a cast here)?

Reply:
I think at the time of the initialization of OnTimerContext, we don't have information of Key type. @robertwb

Reply:
Due to type erasure, we don't have to know it at initialization time, but we do know it when passed to the user (and it'd be better to give them something properly typed.)